pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -107,6 +102,8 @@ let
             # the fake derivations are used for filtering of hyphenation patterns and formats
           else {
             inherit pname version;
+            outPath = "FAKE-PKG-${pname}-${version}"; # required by resolveDeps
```

Comment:
Plans to get rid of this hack? Move it inside resolveDeps?

Reply:
Good point. It is quite possible that replacing
```nix
makeSet = pkgs: { key = (builtins.head pkgs).outPath; inherit pkgs; }
```
with
```nix
makeSet = pkgs: { key = (builtins.head pkgs).outPath or (builtins.head pkgs); inherit pkgs; }
```
is sufficient already. I am not familiar with the internals of `genericClosure`, but I will try and if it works I'll make the change. Maybe a straight `builtins.head pkgs` is enough as it is, actually.

(I suspect I used `outPath` because ages ago I noticed `genericClosure` being considerably faster in a similar setting.)

Reply:
TL;DR: the `key` values must be comparable. But I moved the hack in the `key` itself. Also, I discovered that `outPath` makes `genericClosure` very slow, so I used a shorter string computed from `pname` etc.

Reply:
(Bonus of removing the hack: I figured how to make the code a bit easier, though still not any faster.)